workos/user_management.py

```diff
@@ -1312,120 +1144,118 @@ def enroll_auth_factor(
 
-        Returns: { WorkOSAuthenticationFactorTotp, WorkOSChallenge}
+        Returns: AuthenticationFactorTotpAndChallengeResponse
         """
 
-        if type not in ["totp"]:
-            raise ValueError("Type parameter must be 'totp'")
-
-        headers = {}
-
         payload = {
             "type": type,
             "totp_issuer": totp_issuer,
             "totp_user": totp_user,
             "totp_secret": totp_secret,
         }
 
-        response = self.request_helper.request(
+        response = self._http_client.request(
             USER_AUTH_FACTORS_PATH.format(user_id),
             method=REQUEST_METHOD_POST,
-            headers=headers,
             params=payload,
             token=workos.api_key,
         )
 
-        factor_and_challenge = {}
-        factor_and_challenge["authentication_factor"] = (
-            WorkOSAuthenticationFactorTotp.construct_from_response(
-                response["authentication_factor"]
-            ).to_dict()
-        )
-
-        factor_and_challenge["authentication_challenge"] = (
-            WorkOSChallenge.construct_from_response(
-                response["authentication_challenge"]
-            ).to_dict()
-        )
-
-        return factor_and_challenge
+        return AuthenticationFactorTotpAndChallengeResponse.model_validate(response)
 
     def list_auth_factors(
         self,
-        user_id,
-    ):
+        user_id: str,
+        limit: int = DEFAULT_LIST_RESPONSE_LIMIT,
+        before: Optional[str] = None,
+        after: Optional[str] = None,
+        order: PaginationOrder = "desc",
+    ) -> WorkOsListResource[AuthenticationFactor, AuthenticationFactorsListFilters]:
         """Lists the Auth Factors for a user.
 
         Kwargs:
             user_id (str): The unique ID of the User to list the auth factors for.
 
         Returns:
-            dict: List of Authentication Factors for a User from WorkOS.
+            WorkOsListResource: List of Authentication Factors for a User from WorkOS.
         """
-        response = self.request_helper.request(
+
+        params: ListArgs = {
+            "limit": limit,
+            "before": before,
+            "after": after,
+            "order": order,
+        }
+
+        response = self._http_client.request(
             USER_AUTH_FACTORS_PATH.format(user_id),
             method=REQUEST_METHOD_GET,
+            params=params,
             token=workos.api_key,
         )
 
-        response["metadata"] = {
-            "params": {
-                "user_id": user_id,
-            },
-            "method": UserManagement.list_auth_factors,
+        # We don't spread params on this dict to make mypy happy
+        list_args: AuthenticationFactorsListFilters = {
+            "limit": limit or DEFAULT_LIST_RESPONSE_LIMIT,
+            "before": before,
+            "after": after,
+            "order": order or "desc",
```

Comment:
Previously we changed this to `"order": order` dropping the `or "desc"` part. Should that be the case here?

Reply:
It should, fixed!